features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/prerequisite/SessionReplayRequirement.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.prerequisite
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.utils.submitSafe
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+
+internal class SessionReplayRequirement private constructor(
+    private val checkers: List<SystemRequirementChecker>,
+    private val systemRequirementConfiguration: SystemRequirementConfiguration
+) {
+    private var executorService: ExecutorService = Executors.newSingleThreadExecutor()
+
+    fun runIfMeetRequirements(
+        internalLogger: InternalLogger,
+        runnable: () -> Unit
+    ) {
+        executorService.submitSafe(OPERATION_NAME, internalLogger) {
```

Comment:
are we ok that Session Replay is going to start with delay? And it is going to be initialized on non-main thread.

Reply:
Based on the fact that we need to read some files asynchronously, it seems that we don't have a choice to make it start with a delay. And Me and Jonathan currently don't see a side-effect to register the feature on a non-main thread.

Reply:
> we need to read some files asynchronously before starting

Well, we can still do everything on the main thread and just suppress StrictMode violations (check `StrictModeExt#allowThreadDiskReads`).

These violations don't take into account the amount of penalty, obviously. If I/O on the main thread is caused by very small and fast operation which doesn't add any lag to the FPS, then it is not that critical and may be done.

So it may be violation, but it doesn't cause any significant impact.

But anyway, final decision comes from the profiling data and the architecture.

Reply:
After the discussion with @jonathanmos , we prefer not violate Strict mode here, meaning it's dispatched to a different thread to check the requirements, after it's done, we switch back to main thread to register session replay feature. and `ExecutorService` will be shutdown.

Reply:
As a little trick you can run it on `InternalSdkCore#persistenceExecutor`. The only drawback here is that it may be already many items there to process, but I guess it shouldn't be the case for the app start lifecycle stage (if we assume that Session Replay is initialized there).

Reply:
I passed `sdkCore` inside `runIfMeetRequirements` function to get the persistance executor service.